core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -1930,7 +1929,20 @@ public static MutableRel unifyAggregates(MutableAggregate query,
       }
       final List<AggregateCall> aggregateCalls = new ArrayList<>();
       for (AggregateCall aggregateCall : query.aggCalls) {
-        if (aggregateCall.isDistinct()) {
+        if (aggregateCall.isDistinct() && aggregateCall.getArgList().size() == 1) {
+          int aggIndex = aggregateCall.getArgList().get(0);
+          if (targetGroupByIndexList.contains(aggIndex)) {
+            int newIndex = targetGroupByIndexList.indexOf(aggIndex);
+            aggregateCalls.add(
```

Comment:
if (targetGroupByIndexList.contains(aggIndex)) is not necessary.
You
 can do like this:
```
final int newIndex = targetGroupByIndexList.indexOf(aggIndex);
if (newIndex >=0 ) {
   ......
}
```

Reply:
ok, Thanks.